include/jwt/jwt.hpp

```diff
@@ -147,27 +145,9 @@ struct jwt_set
   {
     using is_transparent = std::true_type;
 
-    bool operator()(const std::string& lhs, const std::string& rhs) const
-    {
-      int ret = strcmp(lhs.c_str(), rhs.c_str());
-      return (ret < 0);
-    }
-
     bool operator()(const jwt::string_view lhs, const jwt::string_view rhs) const
     {
-      int ret = strcmp(lhs.data(), rhs.data());
-      return (ret < 0);
-    }
-
-    bool operator()(const std::string& lhs, const jwt::string_view rhs) const
-    {
-      int ret = strcmp(lhs.data(), rhs.data());
-      return (ret < 0);
-    }
-
-    bool operator()(const jwt::string_view lhs, const std::string& rhs) const
-    {
-      int ret = strcmp(lhs.data(), rhs.data());
+      int ret = lhs.compare(rhs);
```

Comment:
Does std::string get implicitly converted to jwt::string_view ?

Reply:
Yes, to both your custom string_view as well as C++17 string_view. One of the reason string_view was added to the standard is to simplify implementing things like this comparison function here, needing only one function instead of four.